FreeAPS/Sources/Shortcuts/Carbs/CarbPresetIntentRequest.swift

```diff
@@ -0,0 +1,79 @@
+import CoreData
+import Foundation
+
+@available(iOS 16.0,*) final class CarbPresetIntentRequest: BaseIntentsRequest {
+    private var numberFormatter: NumberFormatter {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.maximumFractionDigits = 2
+        return formatter
+    }
+
+    private var presetCarbsList: [carbPresetResult] = []
+
+    override init() {
+        super.init()
+        var carbsList = [Presets]()
+        let requestCarbsList = Presets.fetchRequest() as NSFetchRequest<Presets>
+        try? carbsList = coredataContext.fetch(requestCarbsList)
+        presetCarbsList = carbsList.compactMap {
+            carbPresetResult(
+                id: $0.objectID.uriRepresentation().absoluteString,
+                name: $0.dish ?? "-",
+                carbs: ($0.carbs ?? 0.0) as! Double,
+                fat: ($0.fat ?? 0.0) as! Double,
+                protein: ($0.protein ?? 0.0) as! Double
+            )
+        }
+    }
+
+    func addCarbs(
+        _ quantityCarbs: Double,
```

Comment:
Shouldn't we give these function arguments a proper name? When calling this function you would do `addCarbs(20.0, 10.0, 15.0, '2024-06-06 14:35:22 +0000')`and it might be a bit less error prone to know which parameter is which? Not sure, what do you think? :)

Reply:
Fixed - parameterised